qiskit/providers/ibmq/api_v2/ibmqclient.py

```diff
@@ -64,7 +65,20 @@ def _request_access_token(self):
 
         Returns:
             str: access token.
+        Raises:
+            RequestsApiError: if the user hasn't accepted the license agreement.
         """
+        try:
+            self.client_auth.login(self.api_token)
+        except RequestsApiError as ex:
+            response = ex.original_exception.response
+            status_code = response.status_code
```

Comment:
Can you update the code block to something similar to:
```python
            if response.status_code == 401:
                try:
                    error_code = ...
                    message = ...
                    if error_code == ...:
                        raise AuthenticationLicenseError(ex.original_exception,
                                                         message)
                except (ValueError, KeyError):
                    # A comment saying that the response did not contain the json we were looking for blah
                    pass
```

in order to:
* take advantage that `response.status_code` is always present, so we can enter the `if` early
* prevent the block failing ungracefully via the extra `try` block: there could be cases where the response was not json, or did not contain the keys we expected. By being specific catching those [two](https://2.python-requests.org/en/master/user/quickstart/#json-response-content) [occurences](https://docs.python.org/3/library/exceptions.html#KeyError), we control what happens when that scenario is reached (in this case, we ignore them and let them bubble upwards).
* in this particular case, instead of chaining the exception, "rewrite" it as a `AuthenticationLicenseError` - mostly for compatibility until #106